### PR TITLE
Initial commit of Boolean State Cluster as per issue #227

### PR DESCRIPTION
**Pull request by JimBuzbee**
Hope I'm doing this right!

### CODE REVIEW

File: src/matter/cluster/BooleanStateCluster.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Attributes and commands for Illuminance Measurement.
+ *
+ * @see {@link MatterApplicationClusterSpecificationV1_0} § 1.7
+ */
+export const BooleanStateCluster = Cluster({
+    id: 0x0045,
+    name: "BooleanState",
+    revision: 1,
+    features: { }, // no features
```
Comment (mfucci) [inline src/matter/cluster/BooleanStateCluster.ts:19]: nit: features is optional so you don't need to define this parameter is there are no features

File: src/matter/cluster/BooleanStateCluster.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        * For example, in a Contact Sensor device type, FALSE=open or no contact, TRUE=closed or contact */
+       stateValue: Attribute(0x00, TlvBoolean),
+    },
+        /** @see {@link MatterApplicationClusterSpecificationV1_0 1.7.5} */
+    events: {
+        /** generated when the StateValue attribute changes */
+        stateChange: OptionalEvent(0, EventPriority.Info, { stateValue: TlvField(0, TlvBoolean)  }),
+    },
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.7.6 */
```
Comment (mfucci) [inline src/matter/cluster/BooleanStateCluster.ts:33]: This section doesn't exist in the specs...
You can delete "commands: { }," if there are no commands, this is an optional parameter

**Review by mfucci — DISMISSED** [review]

File: src/matter/cluster/BooleanStateCluster.ts (hunk 2 above)
Comment (JimBuzbee) [reply src/matter/cluster/BooleanStateCluster.ts]: Thanks - Fixed and pushed

File: src/matter/cluster/BooleanStateCluster.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * Attributes and commands for Boolean State.
+ *
+ * @see {@link MatterApplicationClusterSpecificationV1_0} § 1.7
+ */
+export const BooleanStateCluster = Cluster({
+    id: 0x0045,
+    name: "BooleanState",
+    revision: 1,
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.7.5 */
```
Comment (Apollon77) [inline src/matter/cluster/BooleanStateCluster.ts:20]:
```suggestion
    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.7.4 */
```

File: src/matter/cluster/BooleanStateCluster.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    attributes: {
+       /** The semantics of this boolean state are defined by the device type using this cluster.
+        * For example, in a Contact Sensor device type, FALSE=open or no contact, TRUE=closed or contact */
```
Comment (Apollon77) [inline src/matter/cluster/BooleanStateCluster.ts:23]: If a one liner is not posible then lets format the comment correctly

```suggestion
       /** 
        * The semantics of this boolean state are defined by the device type using this cluster.
        * For example, in a Contact Sensor device type, FALSE=open or no contact, TRUE=closed or contact 
        */
```

**Review by Apollon77 — CHANGES_REQUESTED** [review]
Last nits :-) You can just accept the suggetsions. Then I think we can merge. Thank you!

**Comment by JimBuzbee** [comment]
Will start pushing other Clusters today. Forgot to mention, tested this Cluster on Google Home using the "Contact Sensor" device. Seems to work fine

**Review by Apollon77 — APPROVED** [review]

**Comment by Apollon77** [comment]
Thank you and welcome to the contributors :-)

**Comment by JimBuzbee** [comment]
Thanks for putting up with my fumbling :-) Before this, I'd never written a line of Typescript and am still a git novice, but I'll do what I can.

**Pull request merged**